docs/content/topics/scripting/guide.md

```diff
@@ -9,1401 +9,867 @@ aliases:
-```
-
-Note that the `**** AFTER EVENT called` message is printed right before the shutdown
-begins.
+See the [Events] doc for more info on how events are structured in Brigade.
 
-The `error` event is similar, but it is _only_ triggered when a script encounters an
-error.
-
-The `after` and `error` events give you an opportunity to do some final processing (like
-maybe sending a message to your chat server) before exiting.
-
-At this point, we've taken a look at Brigade's event system. But we haven't really done
-anything with Brigade except log a few messages. Let's turn out attention toward Brigade's
-ability to "script up" containers.
+[Events]: /topics/project-developers/events
+[event summary]: /topics/project-developers/events#summary
 
 ## Jobs and Containers
 
-A typical Brigade script divides up the workload like this:
+A typical Brigade script divides up the workload for handling events like this:
 
-- An _event handler_ declares how an event (`push`, `exec`) is processed
-- When Brigade triggers an event, it creates a new `build`, which you can think of as "a
-  specific run of your `brigade.js` file.
-- A build has several _jobs_, where each job starts a container
-- A job runs zero or more _tasks_ inside of a container
+- An _event handler_ declares how an event (`brigade.sh/github:push`,
+  `brigade.sh/cli:exec`) is processed
+- When Brigade triggers an event that a project subscribes to, Brigade creates
+  a _Worker_ per the project's configuration
+- A Worker may run several _Jobs_, where each job starts a container
+- A Job runs zero or more _commands_ inside of a container
 
 And in the next section, we'll talk about how we can _group_ jobs.
 
-In the last section, we focused on the event handlers, and we ran several builds that just
-logged messages. In this section, we'll create a few jobs.
+In the previous section, we focused on the event handlers, which just logged
+messages. In this section, we'll update the event handler to create a few jobs.
 
 To start with, let's create a simple job that doesn't really do any work:
 
 ```javascript
-const { events, Job } = require("brigadier")
+const { events, Job } = require("@brigadecore/brigadier");
 
-events.on("exec", () => {
-  var job = new Job("do-nothing", "alpine:3.4")
+events.on("brigade.sh/cli", "exec", async event => {
+  let job = new Job("do-nothing", "debian", event);
+  await job.run();
+});
 
-  job.run()
-})
+events.process();
 ```
-[brigade-05.js](../../examples/brigade-05.js)
+[03-first-job](https://github.com/brigadecore/brigade/tree/v2/examples/03-first-job)
 
-The first thing to note is that _we have changed the first line_. In addition to importing
-the `events` object, we are now also importing `Job`. `Job` is the prototype for creating
-all jobs in Brigade.
+The first thing to note is that _we have changed the first line_. In addition
+to importing the `events` object, we are now also importing `Job`. `Job` is the
+prototype for creating all jobs in Brigade.
```

Comment:
Nit: As someone who confessed to knowing just enough JS to be dangerous, "prototype" is a JS term that immediately triggers something negative in me. If there's another way to word this, it might be more welcoming to people who aren't so keen on JS.